### PR TITLE
feature(createClient): Add Contentful User Agent Header

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
### TODO

- [x] Add test
- [x]  Update index.d.ts
- [ ] Squash and merge

### CODE REVIEW

**Comment by googol** [comment]
This requires a change to `index.d.ts` and the documentation as well

**Comment by Khaledgarbaya** [comment]
@googol you are the guardian angel of the type definition. I added the 2 new params

File: index.d.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ export interface CreateClientParams {
     httpsAgent?: any;
     proxy?: AxiosProxyConfig;
     headers?: any;
+    application: string;
```
Comment (googol) [inline index.d.ts]: You'll want to add question marks after the names of application and integration, to make these optional.

File: lib/contentful.js [hunk 2]
```diff
@@ -7,11 +7,10 @@
 
 import assign from 'lodash/assign'
 import axios from 'axios'
-import version from '../version'
 import {createHttpClient} from 'contentful-sdk-core'
 import createContentfulApi from './create-contentful-api'
 import createLinkResolver from './create-link-resolver'
-
+import {getUserAgentHeader} from './utils/user-agent'
```
Comment (axe312ger) [inline lib/contentful.js]: It might be smart do add this function as a generic one to `contentful-sdk-core`?

**Comment by axe312ger** [comment]
Still some errors with edge I think

**Comment by Khaledgarbaya** [comment]
@axe312ger I fixed the Tests

**Pull request merged**